bin/activate.bat

```diff
@@ -1,68 +1,54 @@
 @echo off
+setlocal
 
-for /f "delims=" %%i in ("%~dp0..\envs") do (
-    set ANACONDA_ENVS=%%~fi
-)
-
-set CONDA_NEW_NAME=%~1
+set CONDA_NEW_ENV=%1
+set CONDA_NEW_ENV=%CONDA_NEW_ENV:"=%
 
-if "%~2" == "" goto skiptoomanyargs
-    echo ERROR: Too many arguments provided
-    goto usage
+if "%2" == "" goto skiptoomanyargs
+    (echo Error: did not expect more than one argument.) 1>&2
+    exit /b 1
 :skiptoomanyargs
 
-if "%CONDA_NEW_NAME%" == "" set CONDA_NEW_NAME=%~dp0..\
-
-if exist "%ANACONDA_ENVS%\%CONDA_NEW_NAME%\conda-meta" goto usenamedenv
-    for /F %%i in ("%CONDA_NEW_NAME%") do set CONDA_NEW_PATH=%%~fi
-    if exist "%CONDA_NEW_PATH%\conda-meta" goto usefullpath
-        echo No environment named "%CONDA_NEW_NAME%" exists in %ANACONDA_ENVS%
-        set CONDA_NEW_NAME=
-        set CONDA_NEW_PATH=
-        exit /b 1
-:usenamedenv
-    set CONDA_NEW_PATH=%ANACONDA_ENVS%\%CONDA_NEW_NAME%
-    goto skipmissingenv
-:usefullpath
-    for /F %%i in ("%CONDA_NEW_PATH%") do set CONDA_NEW_NAME=%%~ni
-:skipmissingenv
+if not "%1" == "" goto skipmissingarg
+    (echo Error: no environment provided.) 1>&2
+    exit /b 1
+:skipmissingarg
+
+if not "%1" == "--help" goto skipusage
+    (
+    echo Usage: activate ENV
+    echo.
+    echo Deactivates previously activated Conda
+    echo environment, then activates the chosen one.
+    ) 1>&2
+    exit /b 1
+:skipusage
+
+REM Use conda itself to figure things out
+REM No conda, no dice
```

Comment:
If it makes sense, we can just point it to the same folder as activate.bat to look for conda.exe.  Conda can only be installed into a root environment, and should be easily locatable using a relative path given the path of activate.bat